app/controllers/api/databases.php

```diff
@@ -3649,8 +3649,20 @@ function updateAttribute(
 
             $cursor->setValue($cursorDocument);
         }
+
+        $selectQueries = [];
+
         try {
-            $documents = $dbForProject->find('database_' . $database->getSequence() . '_collection_' . $collection->getSequence(), $queries);
+            $selectQueries = Query::groupByType($queries)['selections'] ?? [];
+
+            if (! empty($selectQueries)) {
+                // has selects, allow relationship on documents!
+                $documents = $dbForProject->find('database_' . $database->getSequence() . '_collection_' . $collection->getSequence(), $queries);
+            } else {
+                // has no selects, disable relationship looping on documents!
+                $documents = $dbForProject->skipRelationships(fn () => $dbForProject->find('database_' . $database->getSequence() . '_collection_' . $collection->getSequence(), $queries));
```

Comment:
I think we still can't merge because of this, relationships won't be fetched without passing select queries

Reply:
@abnegate lets mark this PR in draft then? removing this check basically makes the PR of not much use anyways. We can merge it in the end since the structure would be the module by the end anyways. Wdyt?